canonical/checkbox
# Add a new function to find which renderer is running the process (New)

## Description


While testing without pci id, user couldn't know the process is running
on which renderer. Therefore, this change is going to:
* add a new function to get which GPU is executing the command that could help user to make sure the renderer is not llvmpipe.
* refactor some codes to make reuse easier.
* add PCI ID in the log to make verify easier.

The jobs.pxu and test-plan.pxu will be in another PR, and the draft jobs and test-plan that use this new function as below:

```
plugin: shell
category_id: com.canonical.plainbox::graphics
id: graphics/auto_glxgears
flags: also-after-suspend
user: root
requires:
    executable.name == 'glxgears'
    dmi.product in ['Desktop','Low Profile Desktop','Tower','Mini Tower','Space-saving']
command:
    prime_offload_tester.py -c glxgears -t 30
summary:
    Test that glxgears works for current video card
purpose:
     Tests the basic 3D capabilities of your current video card. This test covers all devices without an integrated display, such as desktops.

plugin: shell
category_id: com.canonical.plainbox::graphics
id: graphics/auto_glxgears_fullscreen
flags: also-after-suspend
user: root
requires:
    executable.name == 'glxgears'
    dmi.product in ['Desktop','Low Profile Desktop','Tower','Mini Tower','Space-saving']
command:
     prime_offload_tester.py -c "glxgears -fullscreen" -t 30
summary:
    Test that glxgears works in full screen mode for current video card
purpose:
     Tests the basic full screen 3D capabilities of your current video card. This test covers all devices without an integrated display, such as desktops.
plugin: user-interact-verify
category_id: com.canonical.plainbox::graphics
id: graphics/valid_glxgears
flags: also-after-suspend
user: root
requires:
    executable.name == 'glxgears'
    dmi.product in ['Desktop','Low Profile Desktop','Tower','Mini Tower','Space-saving']
command:
     prime_offload_tester.py -c glxgears -t 30
summary:
    Test that glxgears works for current video card
purpose:
     Tests the basic 3D capabilities of your current video card. This test covers all devices without an integrated display, such as desktops.
steps:
     1. Click "Test" to execute an OpenGL demo. Press ESC at any time to close.
     2. Verify that the animation is not jerky or slow.
verification:
     1. Did the 3d animation appear?
     2. Was the animation free from slowness/jerkiness?

plugin: user-interact-verify
category_id: com.canonical.plainbox::graphics
id: graphics/valid_glxgears_fullscreen
flags: also-after-suspend
user: root
requires:
    executable.name == 'glxgears'
    dmi.product in ['Desktop','Low Profile Desktop','Tower','Mini Tower','Space-saving']
command:
     prime_offload_tester.py -c "glxgears -fullscreen" -t 30
summary:
    Test that glxgears works in full screen mode for current video card
purpose:
     Tests the basic full screen 3D capabilities of your current video card. This test covers all devices without an integrated display, such as desktops.
steps:
     1. Click "Test" to execute an OpenGL demo. Press ESC at any time to close.
     2. Verify that the animation is not jerky or slow.
verification:
     1. Did the 3d animation appear?
     2. Was the animation free from slowness/jerkiness?
```

```
id: graphics-gpu-cert-automated
unit: test plan
_name: Graphics tests (Automated)
_description:
 Graphics tests (Automated)
include:
~ skip ~
 graphics/auto_glxgears                         certification-status=blocker
 graphics/auto_glxgears_fullscreen              certification-status=blocker
~ skip ~
bootstrap_include:
    graphics_card

id: graphics-gpu-cert-manual
unit: test plan
_name: Graphics tests (Manual)
_description:
 Graphics tests (Manual)
include:
~ skip ~
 graphics/valid_glxgears                        certification-status=blocker
 graphics/valid_glxgears_fullscreen             certification-status=blocker
~ skip ~
bootstrap_include:
    graphics_card
```

## Resolved issues



## Documentation



## Tests


Unit test and
laptop: [auto](https://certification.canonical.com/hardware/202001-27667/submission/369177/), [manual](https://certification.canonical.com/hardware/202001-27667/submission/369178/)
desktop: [auto](https://certification.canonical.com/hardware/202401-33394/submission/369179/), [manual](https://certification.canonical.com/hardware/202401-33394/submission/369181/)
